src/Entities/PaymentInstrumentValidation.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * This source file is proprietary and part of Rebilly.
+ *
+ * (c) Rebilly SRL
+ *     Rebilly Ltd.
+ *     Rebilly Inc.
+ *
+ * @see https://www.rebilly.com
+ */
+
+namespace Rebilly\Entities;
+
+use Rebilly\Rest\Resource;
+use DomainException;
+
+abstract class PaymentInstrumentValidation extends Resource
+{
+    protected static $supportMethods = [
+        PaymentMethod::METHOD_PAYMENT_CARD,
+    ];
+
+    /**
+     * @param array $data
+     *
+     * @return PaymentCardValidation
+     */
+    public static function createFromData(array $data)
+    {
+        if (!isset($data['method'])) {
+            throw new DomainException('Method is required');
+        }
+        switch ($data['method']) {
+            case PaymentMethod::METHOD_PAYMENT_CARD:
+                $paymentInstrumentValidation = new PaymentCardValidation($data);
+
+                break;
+            default:
+                throw new DomainException(
+                    sprintf('Unexpected method. Only %s methods support', implode(',', self::$supportMethods))
+                );
+        }
+
+        return $paymentInstrumentValidation;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return $this
+     */
+    public function setMethod($value)
```

Comment:
This setter should not be accessible, because it break the object. Imagine you have instance of the `PaymentCardValidation` and change method to invalid value.